src/app.tsx

```diff
@@ -183,6 +184,14 @@ export default class App extends React.Component<{}, IState> {
                                 theme="dark"
                                 onNavigationStateChange={(_, newState) => {
                                     this.setState({ navigationState: newState });
+
+                                    const currentRoute = NavigationService.getCurrentRouteWithParams();
```

Comment:
we could keep current route object at app level, and if it's changed we trigger addBreadcrumb, if not, we do nothing